policy/predicate/predicate.go

```diff
@@ -27,3 +30,61 @@ type Predicate interface {
 	// Evaluate determines if the predicate is satisfied.
 	Evaluate(ctx context.Context, prctx pull.Context) (*common.PredicateResult, error)
 }
+
+type unit struct{}
+type set map[string]unit
+
+// allowedConclusions can be one of:
+// action_required, cancelled, failure, neutral, success, skipped, stale,
+// timed_out
+type allowedConclusions set
```

Comment:
Curious for your opinion here: there's a decent amount of code in this PR for converting between this set representation and a list of strings in different circumstances. Given that the number of elements in the set is almost always going to be less than 3, what do you think about using a simple `[]string` and testing for matching conclusions with `slices.Contains`?

That could allow duplicates, but duplicates would be obvious in the policy definition and shouldn't cause any problems (other than slightly awkward messages.)

Reply:
Yeah no worries. I'm used to writing such code so I tend to reach for it but it's fine to skip it in this case. See ebee53203ad74705387c4abe080d22ce31fe7f4e.

If needed we could dedupe when printing, when looking at the slice or any other time. But I doubt it'll be a big deal since we're talking about a small number of possible values.